Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1669 @@
+                }
+            }
+
+            set
+            {
+                if (HasVBEExtensions)
+                {
+                    var rawArgsString = string.Join(" : ", value.Select(x => x.Key + " = " + x.Value));
+                    ConditionalCompilationArgumentsRaw = rawArgsString;
+                }
+                else
+                {
+                    throw new ArgumentException("This ITypeLib is not hosted by the VBE, so does not support ConditionalCompilationArguments");
+                }
+            }
+        }
+        
+        public void Document(StringLineBuilder output)
+        {
+            output.AppendLine();
+            output.AppendLine("================================================================================");
+            output.AppendLine();
+
+            var libName = Name ?? "[VBA.Immediate.Window]";     
+
+            output.AppendLine("ITypeLib: " + Name);
+            output.AppendLineNoNullChars("- Documentation: " + DocString);
+            output.AppendLineNoNullChars("- HelpContext: " + HelpContext);
+            output.AppendLineNoNullChars("- HelpFile: " + HelpFile);
+            output.AppendLine("- Guid: " + Attributes.guid);
+            output.AppendLine("- Lcid: " + Attributes.lcid);
+            output.AppendLine("- SysKind: " + Attributes.syskind);
+            output.AppendLine("- LibFlags: " + Attributes.wLibFlags);
+            output.AppendLine("- MajorVer: " + Attributes.wMajorVerNum);
+            output.AppendLine("- MinorVer: " + Attributes.wMinorVerNum);
+            output.AppendLine("- HasVBEExtensions: " + HasVBEExtensions);
+            if (HasVBEExtensions)
+            {
+                output.AppendLine("- VBE Conditional Compilation Arguments: " + ConditionalCompilationArguments);
+
+                foreach (var reference in VBEReferences)
+                {
+                    reference.Document(output);
+                }
+            }
+
+            output.AppendLine("- TypeCount: " + TypesCount);
+
+            foreach (var typeInfo in TypeInfos)
+            {
+                using (typeInfo)
+                {
+                    typeInfo.Document(output, libName, 0);
+                }
+            }
+        }
+
+        // We have to wrap the ITypeInfo returned by GetTypeInfoOfGuid
+        // so we cast to our IVBETypeLib interface in order to work with the raw IntPtr for aggregation
+        void ComTypes.ITypeLib.GetTypeInfoOfGuid(ref Guid guid, out ComTypes.ITypeInfo ppTInfo)
+        {
+            IntPtr typeInfoPtr = IntPtr.Zero;
+            ((ITypeLib_Ptrs)target_ITypeLib).GetTypeInfoOfGuid(guid, out typeInfoPtr);
+            var outVal = new TypeInfoWrapper(typeInfoPtr);  // takes ownership of the COM reference
+            ppTInfo = outVal;
+
+            _typeInfosWrapped = _typeInfosWrapped ?? new DisposableList<Ty
```

Comment:
This is the line I meant to point out in my prior comment. Hopefully it gets pinned to correct line this time. 😕

Reply:
thanks Ben, not sure what's wrong with the pinning there.  Anyway, the Dispose is still required (as was the other one you mentioned), due to the implicit IDisposable in IEnumerable, but sure the explicit IDisposable has now been removed.  Thanks.